### ISSUE TITLE
optional gzip compression on save files

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#436]
It is beneficial to have an option in the `Save` statement in such a way that FORM can gzip-compress expressions in save files. See also the discussion at https://github.com/vermaseren/form/pull/433.

A possible way for the syntax is:
```
Save gzip savefile;
Save gzip savefile expressions...;
```
Here `gzip` is a contextual keyword. The disadvantage of this syntax is that the user cannot specify `gzip` as the save filename (but normally nobody would want to do that).

If the save file format has room to record whether expressions in the file are gzip-compressed or not  (or with other compression formats), then the syntax of the `Load` statement does not need to be changed; the user can just use `Load` as in the past and it transparently loads expressions if compressed.

**Comment by vermaseren** [comment]
I guess the best is to make it
Save -gzip savefile etc;
It would be a very bad habit to start a filename with -
That can really get things messed up.

To figure out all necessary details and possibly implementing this would be a very
good exercise for the workshop in April.

**Comment by tueda** [comment]
We can also use `+gzip`, for consistency with `+s` and `-s` options etc. of the `Print` statement (`+gzip`: enabled, `-gzip`: disabled).

**Comment by vermaseren** [comment]
Yes, if for + holds the same as for - in running shell commands and hence makes the
use of filenames that start with + extremely dangerous. There are indeed optionsin commands
that start with + as in chmod.
